packages/frontend/src/features/semanticViewer/runners/SemanticViewerResultsRunner.ts

```diff
@@ -3,37 +3,23 @@ import {
     DimensionType,
     FieldType,
     SemanticLayerFieldType,
+    VizAggregationOptions,
     VizIndexType,
     VIZ_DEFAULT_AGGREGATION,
     type IResultsRunner,
     type PivotChartData,
     type RawResultRow,
+    type RunPivotQuery,
     type SemanticLayerField,
-    type SemanticLayerPivot,
     type SemanticLayerQuery,
-    type SemanticViewerPivotChartLayout,
-    type SqlRunnerPivotChartLayout,
     type VizColumn,
     type VizCustomMetricLayoutOptions,
     type VizIndexLayoutOptions,
     type VizValuesLayoutOptions,
 } from '@lightdash/common';
 import { apiGetSemanticLayerQueryResults } from '../api/requests';
 
-const transformChartLayoutToSemanticPivot = (
-    config: SemanticViewerPivotChartLayout,
-): SemanticLayerPivot => {
-    if (!config.x) {
-        throw new Error('X is required');
-    }
-
-    return {
-        on: [config.x.reference],
-        index: config.groupBy?.map((groupBy) => groupBy.reference) ?? [],
-        values: config.y.map((y) => y.reference),
-    };
-};
-
+// not useful - semantic layer field type should be source of truth
```

Comment:
Almost all of these "conversion functions" are needed because we don't have unified types across both runners, let's make it a priority to remove them

Reply:
Agreed. This seems like the hardest part, no?